File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20013Steps.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20013Ingredients
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param number 
 * @param step 
 * @param ingredients 
 * @param equipment 
 */
case class InlineResponse20013Steps(number: BigDecimal,
                step: String,
                ingredients: Option[Seq[InlineResponse20013Ingredients]],
                equipment: Option[Seq[InlineResponse20013Ingredients]]
                )

object InlineResponse20013Steps {
    /**
     * Creates the codec for converting InlineResponse20013Steps from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20013Steps] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20013Steps] = deriveEncoder
}

File: dart/com/spoonacular/client/model/inline_response20013_steps.dart
part of openapi.api;

class InlineResponse20013Steps {
  
  num number = null;
  
  String step = null;
  
  List<InlineResponse20013Ingredients> ingredients = [];
  
  List<InlineResponse20013Ingredients> equipment = [];
  InlineResponse20013Steps();

  @override
  String toString() {
    return 'InlineResponse20013Steps[number=$number, step=$step, ingredients=$ingredients, equipment=$equipment, ]';
  }

  InlineResponse20013Steps.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['number'] == null) {
      number = null;
    } else {
          number = json['number'];
    }
    if (json['step'] == null) {
      step = null;
    } else {
          step = json['step'];
    }
    if (json['ingredients'] == null) {
      ingredients = null;
    } else {
      ingredients = InlineResponse20013Ingredients.listFromJson(json['ingredients']);
    }
    if (json['equipment'] == null) {
      equipment = null;
    } else {
      equipment = InlineResponse20013Ingredients.listFromJson(json['equipment']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (number != null)
      json['number'] = number;
    if (step != null)
      json['step'] = step;
    if (ingredients != null)
      json['ingredients'] = ingredients;
    if (equipment != null)
      json['equipment'] = equipment;
    return json;
  }

  static List<InlineResponse20013Steps> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20013Steps>() : json.map((value) => new InlineResponse20013Steps.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20013Steps> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20013Steps>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20013Steps.fromJson(value));
    }
    return map;
  }
}

